packages/message-box/src/index.vue

```diff
@@ -0,0 +1,369 @@
+      </div>
+    </div>
+  </transition>
+</template>
+<script lang='ts'>
+import {
+  defineComponent,
+  nextTick,
+  onMounted,
+  onBeforeUnmount,
+  computed,
+  watch,
+  watchEffect,
+  getCurrentInstance,
+  reactive,
+  toRefs,
+} from 'vue'
+import { addClass, removeClass } from '@element-plus/utils/dom'
+import ElButton from '@element-plus/button/src/button.vue'
+import ElInput from '@element-plus/input/src/index.vue'
+import { t } from '@element-plus/locale'
+import Dialog  from '@element-plus/utils/aria-dialog'
+import usePopup from '@element-plus/utils/popup/usePopup'
+
+let dialog
+
+const TypeMap: Indexable<string> = {
+  success: 'success',
+  info: 'info',
+  warning: 'warning',
+  error: 'error',
+}
+
+export default defineComponent({
+  name: 'ElMessageBox',
+  components: {
+    ElButton,
+    ElInput,
+  },
+  props: {
+    openDelay: {},
+    closeDelay: {},
+    zIndex: {},
+    modalFade: {
+      type: Boolean,
+      default: true,
+    },
+    modalClass: {
+      type: String,
+      default: '',
+    },
+    modalAppendToBody: {
+      type: Boolean,
+      default: false,
+    },
+    modal: {
+      type: Boolean,
+      default: true,
+    },
+    lockScroll: {
+      type: Boolean,
+      default: true,
+    },
+    showClose: {
+      type: Boolean,
+      default: true,
+    },
+    closeOnClickModal: {
+      type: Boolean,
+      default: true,
+    },
+    closeOnPressEscape: {
+      type: Boolean,
+      default: true,
+    },
+    closeOnHashChange: {
+      type: Boolean,
+      default: true,
+    },
+    center: {
+      default: false,
+      type: Boolean,
+    },
+    roundButton: {
+      default: false,
+      type: Boolean,
+    },
+  } ,
+  setup(props) {
+    let vm
+    const popup = usePopup(props, doClose)
+    const state = reactive<State>({
+      uid: 1,
+      title: undefined,
+      message: '',
+      type: '',
+      iconClass: '',
+      customClass: '',
+      showInput: false,
+      inputValue: null,
+      inputPlaceholder: '',
+      inputType: 'text',
+      inputPattern: null,
+      inputValidator: null,
+      inputErrorMessage: '',
+      showConfirmButton: true,
+      showCancelButton: false,
+      action: '',
+      confirmButtonText: '',
+      cancelButtonText: '',
+      confirmButtonLoading: false,
+      cancelButtonLoading: false,
+      confirmButtonClass: '',
+      confirmButtonDisabled: false,
+      cancelButtonClass: '',
+      editorErrorMessage: null,
+      callback: null,
+      dangerouslyUseHTMLString: false,
+      focusAfterClosed: null,
+      isOnComposition: false,
+      distinguishCancelAndClose: false,
+      type$: '',
+    })
+    const icon = computed(() => state.iconClass || (state.type && TypeMap[state.type] ? `el-icon-${ TypeMap[state.type] }` : ''))
+
+    const confirmButtonClasses = computed(() => `el-button--primary ${ state.confirmButtonClass }`)
+
+    const cancelButtonClasses = computed(() => `${ state.cancelButtonClass }`)
```

Comment:
I'm sorry, I don't quite understand your intension here, you can just use `state.cancelButtonClass` why set it as a `ComputedRef`?

Reply:
i removed it